modules/programs/hexchat.nix

```diff
@@ -0,0 +1,248 @@
+            };
+          in mkOption {
+            type = nullOr (channelOptions);
+            description = "Channel options";
+            example = literalExample
+              "{ autoconnect = true; useGlobalUserInformation = true; }";
+          };
+          password = mkOption {
+            type = str;
+            description = "Password";
+          };
+          realName = mkOption {
+            type = str;
+            description = "Real name";
+          };
+          servers = mkOption {
+            type = (listOf str);
+            description = "IRC Server Address List";
+            example =
+              literalExample ''[ "chat.freenode.net" "irc.freenode.net" ]'';
+            default = [ ];
+          };
+          userName = mkOption {
+            type = str;
+            description = "User name";
+          };
+        };
+      };
+    in mkOption {
+      type = nullOr (attrsOf modChannelOption);
+      description = "Configures '~/.config/hexchat/servlist.conf'";
+      example = literalExample ''
+        {
+          freenode = {
+            autojoin = [
+              "#home-manager"
+              "##linux"
+              "#nixos"
+            ];
+            charset = "UTF-8 (Unicode)";
+            commands = [
+              "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+            ];
+            loginMethod = sasl;
+            nickname = "my_nickname";
+            nickname2 = "my_secondchoice";
+            options = {
+              acceptInvalidSSLCertificates = false;
+              autoconnect = true;
+              bypassProxy = true;
+              connectToSelectedServerOnly = true;
+              useGlobalUserInformation = false;
+              useSSLForAllServers = false;
+            };
+            password = "my_password";
+            realName = "my_realname";
+            servers = [
+              "chat.freenode.net"
+              "irc.freenode.net"
+            ];
+            userName = "my_username";
+          };
+        }'';
+    };
+    settings = mkOption {
+      description = "Configures '~/.config/hexchat/hexchat.conf'";
+      example = literalExample ''
+        {
+          irc_nick1 = "mynick";
+          irc_username = "bob";
+          irc_realname = "Bart Simpson";
+          text_font = "Monospace 14";
+        };'';
+      type = nullOr (attrsOf str);
+    };
+    enable = mkEnableOption "HexChat - Graphical IRC client";
+    theme = mkOption {
+      description =
+        "Theme package for HexChat. Expects a derivation including decompressed theme files. '.hct' format requires unzip decompression.";
+      type = nullOr package;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.hexchat ];
+    xdg.configFile."hexchat" = mkIf (cfg.theme != null) {
+      source = cfg.theme;
+      recursive = true;
+    };
+    xdg.configFile."hexchat/hexchat.conf" = mkIf (cfg.settings != null) {
+      force = true;
```

Comment:
https://github.com/nix-community/home-manager#keeping-your--safe-from-harm

Reply:
Are you debating the need of 'force'? Force is required because HexChat will replace hexchat.conf everytime it is closed. And it does it, in reloading home-manage it require manually removing `hexchat.conf`, which would be pretty annoying.

Reply:
I could turn this into a flag like `deleteHexchatConfigFilesAutomatically` but this feature will be necessary every time.

Reply:
Ah ok, this makes sense.

But you need to document those things better. I think adding a comment here saying that HexChat is deleting and recreating this file everytime it runs (this is the only way I can think for this behavior, since `/nix/store` files are immutable).

Also, probably some warning to be given to the user since it can potentially lose data. I don't know the best way to do it, probably @rycee can give some tips.

Reply:
Also, considering this warning from README: https://github.com/nix-community/home-manager#words-of-warning

Seems those cases are expected, so maybe just removing `force` and assuming that this module is mutable? An option doesn't seem a bad idea either...